resources/js/package.js

```diff
@@ -1,37 +1,69 @@
-document.addEventListener('turbo:load', function () {
-    ['shipping_address', 'billing_address'].forEach((type) => {
-        window.app.$watch('checkout.'+type+'.postcode', function () {
-            getAddressFromPostcodeservice(type)
-        })
+import { useThrottleFn } from "@vueuse/core"
+
+const postcodeObserver = new MutationObserver(useThrottleFn(() => {
+    let types = ['housenumber', 'postcode', 'country']
+    types.forEach((type) => {
+        document.querySelectorAll(`[id\$='_${type}']`).forEach(el => {
+            if (el.id in window.app.custom.postcode_data) {
+                return
+            }
 
-        window.app.$watch('checkout.'+type+'.street.1', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.custom.postcode_data[el.id] = el.value
+            let id = el.id.replace(`_${type}`, '')
+            el.addEventListener('change', () => {
```

Comment:
I wouldn't use addEventListener here as any change in the dom tree (including adding/removing classes dynamically, changing the value of an input) would trigger a new event listener to be added.

UseEventListener would be one step, but we need to track if an element already has an event listener this way

Reply:
In this case it only happens once because of the check above: it puts `el.id` into the `postcode_data` variable, and then checks for that on the next pass.

I double checked with how Turbo handles it and it doesn't have any issues with it.